source/solvers/heat_transfer_assemblers.cc

```diff
@@ -421,8 +423,10 @@ HeatTransferAssemblerRobinBC<dim>::assemble_matrix(
                               const double phi_face_T_j =
                                 scratch_data.phi_face_T[f][q][j];
                               local_matrix(i, j) +=
-                                (h + 4.0 * Stefan_Boltzmann_constant *
-                                       emissivity * T_face * T_face * T_face) *
+                                (h +
+                                 4.0 * Stefan_Boltzmann_constant * emissivity *
+                                   T_face * T_face * T_face +
+                                 heat_source / T_face) *
```

Comment:
Why are you dividing by T_face
A heat flux is a constant flux applied to a face. You should not divide by the temperature.
This is not temperature dependant